### PR TITLE
Delete unnecessary files and fix format in some MD pages

### PR DESCRIPTION
**Pull request by rchildre3**
* Bolster the `.gitignore`
* Follow `CONTRIBUTING.md` guidelines and wrap `README.md` to 80
  characters and separate links for readability
* Remove pesky Unicode dash in `CONTRIBUTING.md`
* Remove `.DS_STORE` files
* Rename `#project.yaml#` to something logical and enable syntax
  highlighting

### CODE REVIEW

**Comment by rchildre3** [comment]
@googlebot I signed it!

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,456 @@
-.vscode/
-*.pyc
-/build/
-*~
\ No newline at end of file
+# Created by https://www.toptal.com/developers/gitignore/api/python,c,c++,pycharm,visualstudiocode,vim,emacs,linux,windows,macos
```
Comment (Dor1s) [inline .gitignore:1]: please revert changes in this file, although adding `.DS_Store` in here would be helpful, but no need to put this large auto-generated one

Comment (inferno-chromium) [reply .gitignore:1]: +1

File: README.md [hunk 2]
```diff
@@ -1,40 +1,58 @@
...
+techniques with scalable, distributed execution.
+
+[Core Infrastructure Initiative]: https://www.coreinfrastructure.org/
 
-In cooperation with the [Core Infrastructure Initiative](https://www.coreinfrastructure.org/), 
-OSS-Fuzz aims to make common open source software more secure and stable by
-combining modern fuzzing techniques with scalable,
-distributed execution.
+We support the [libFuzzer], [AFL], and [Honggfuzz] fuzzing engines in
+combination with [Sanitizers], as well as [ClusterFuzz]; a distributed fuzzer
```
Comment (inferno-chromium) [inline README.md:22]: [ClusterFuzz],

File: README.md [hunk 3]
```diff
@@ -1,40 +1,58 @@
 # OSS-Fuzz: Continuous Fuzzing for Open Source Software
```
Comment (inferno-chromium) [inline README.md:1]: Please also modify https://github.com/google/oss-fuzz/blob/master/docs/index.md in a similar way. it is the same content.

**Comment by rchildre3** [comment]
Would the reviewers be willing to accept a much smaller `.gitignore`?

The new file is far smaller and I think truly should be in the file to prevent such spurious files like `.DS_STORE` from being committed again

**Comment by inferno-chromium** [comment]
> Would the reviewers be willing to accept a much smaller `.gitignore`?
> 
> The new file is far smaller and I think truly should be in the file to prevent such spurious files like `.DS_STORE` from being committed again

Smaller and only things that could be applicable.

File: README.md [hunk 4]
```diff
@@ -1,40 +1,58 @@
...
+[Honggfuzz]: https://github.com/google/honggfuzz
+[Sanitizers]: https://github.com/google/sanitizers
+[ClusterFuzz]: https://github.com/google/clusterfuzz
 
-Currently, OSS-Fuzz supports C/C++, Rust, and Go code. Other languages supported by [LLVM](http://llvm.org) may work too.
-OSS-Fuzz supports fuzzing x86_64 and i386 builds.
+Currently, OSS-Fuzz supports C/C++, Rust, and Go code. Other languages supported
+by [LLVM] may work too. OSS-Fuzz supports fuzzing x86_64 and i386 builds.
+
+[LLVM]: http://llvm.org
```
Comment (Dor1s) [inline README.md:34]: sorry for another nit, since we're here, could you please use https for this link

File: docs/index.md [hunk 5]
```diff
@@ -9,42 +9,53 @@ has_toc: false
...
-[Honggfuzz](https://github.com/google/honggfuzz) fuzzing engines
-in combination with [Sanitizers](https://github.com/google/sanitizers), as well as
-[ClusterFuzz](https://github.com/google/clusterfuzz),
-a distributed fuzzer execution environment and reporting tool. 
+Currently, OSS-Fuzz supports C/C++, Rust, and Go code. Other languages supported
+by [LLVM] may work too. OSS-Fuzz supports fuzzing x86_64 and i386 builds.
 
-Currently, OSS-Fuzz supports C/C++, Rust, and Go code. Other languages supported by [LLVM](http://llvm.org) may work too.
-OSS-Fuzz supports fuzzing x86_64 and i386 builds.
+[LLVM]: http://llvm.org
```
Comment (Dor1s) [inline docs/index.md:43]: https

File: README.md (hunk 4 above)
Comment (rchildre3) [reply README.md]: resolved in 6015ee628a1034346128bde195a58cd139ea5d6b

File: docs/index.md (hunk 5 above)
Comment (rchildre3) [reply docs/index.md]: resolved in 6015ee628a1034346128bde195a58cd139ea5d6b

**Pull request merged**